### PR TITLE
Preliminary Store Controller Setup

### PR DESCRIPTION
**Pull request by BitterChris**
## Purpose
Adds StoreController and implements the index action which displays all Products ordered by ascending title (alphabetically).

## How
- Generator to create controller and setup index method.
- Redesign application layout and implement new stylesheet.

## Divergences
- Added `pry` gem for easier debugging.
- Instead of hardcoding min-width values for media queries, I used a variable instead.

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          background: none;
+          color: white;
+        }
+      }
+    }
+  }
+  main {
+    padding: 0.5em;
+  }
+}
```
Comment (jesseschroder) [inline app/assets/stylesheets/application.scss]: eof line

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,8 +10,22 @@
...
-      <%= yield %>
-    </main>
+    <header class="main">
+      <%= image_tag 'logo.svg', alt: 'The Pragmatic Bookshelf' %>
+      <h1><%= @page_title %></h1>
+    </header>
+    <section class="content">
+      <nav class="side_nav">
+        <ul>
+          <li><a href="/">Home</a></li>
```
Comment (jesseschroder) [inline app/views/layouts/application.html.erb]: Are `<a>` tags preferred here over rails `link_to`?

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (BitterChris) [reply app/assets/stylesheets/application.scss]: ffs

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (BitterChris) [reply app/views/layouts/application.html.erb]: Oh yeah good call. That was a flat out verbatim re-write of what the book says, but I like `link_to` much more.

**Pull request merged**